whatshap/vcf.py

```diff
@@ -963,7 +984,18 @@ def write(self, chromosome: str, sample_superreads, sample_components):
                         )
                         is_het = not genotypes[pos].is_homozygous()
 
-                    if pos in components and pos in phases and is_het:
+                    if (
```

Comment:
A couple of comments earlier, I made a suggestion of how to avoid the duplicate call to `VcfWriter.write`. I suggest to do something similar here: Call `self._set_phasing_tags` only once with four parameters, but use an `if` before that call to set the fourth parameter.

Also, list the `pos in components and pos in phases and is_het` conditionals only once. The `if haploid_components ...` can be in a nested if. (Not sure if that explanation is sufficient to understand what I mean, just let me know.)